rest-core/src/Rest/Driver/Routing.hs

```diff
@@ -80,24 +80,24 @@ runRouter method uri = runIdentity
                      . flip runReaderT method
                      . unRouter
 
-route :: Maybe Method -> UriParts -> Rest.Api m -> Either Reason_ (RunnableHandler m)
-route Nothing       _   _   = apiError UnsupportedMethod
-route (Just method) uri api = runRouter method uri $
+route :: Config m -> Maybe Method -> UriParts -> Rest.Api m -> Either Reason_ (RunnableHandler m)
+route _    Nothing      _   _   = apiError UnsupportedMethod
+route cfg (Just method) uri api = runRouter method uri $
   do versionStr <- popSegment
      case versionStr `Rest.lookupVersion` api of
-          Just (Some1 router) -> routeRoot router
+          Just (Some1 router) -> routeRoot cfg router
           _                   -> apiError UnsupportedVersion
 
-routeRoot :: Rest.Router m s -> Router (RunnableHandler m)
-routeRoot router@(Rest.Embed resource _) = do
+routeRoot :: Config m -> Rest.Router m s -> Router (RunnableHandler m)
```

Comment:
I'd still add the config inside the reader in `Router` if possible. Or did that have issues?